siliconcompiler/schema_support/option.py

```
@@ -1,8 +1,431 @@
+        schema.insert(
+            'maxthreads',
+            Parameter(
+                'int',
+                scope=Scope.GLOBAL,
+                shorthelp="Option: maximum number of threads to assign a task",
+                example=["api: option.set('maxthreads', 4)"],
+                help="""
+                Maximum number of threads for each task in a job. If not set this will default
+                to the number of cpu cores available."""))
+
+    def get_name(self, step: str = None, index: str = None) -> str:
```

Ruff (0.13.2)

PEP 484 prohibits implicit `Optional`

Convert to `Optional[T]`

(RUF013)